### PR TITLE
1.4.2 Update Rework/Refactor/Compatibility

### PR DESCRIPTION
**Pull request by fonnymunkey**
TL;DR:
Rework to patch multiple bugs/incompatibilities/performance, optional compatibilities added such as Ice and Fire (Flamed/Iced weapons), Inspirations (Cauldron compat), etc. Hard dependency of ReachFix added to properly handle spear and nunchaku attack targeting and reach in a way that is compatible with other mods.

More comprehensive list below, I'm sure I missed some as the list of changes is big, so far everything is tested and seems to be working properly, I'm sure issues will crop up as public testing begins, both pre-existing and newly introduced bugs.

- Updated to FG5/Gradle 7.5, reworking gradle to use gradle.properties, fixed gradle unable to build.
- Removed custom reach packet handling and manual nunchaku targetting spamming packets, moved to use ReachFix instead.
- Removed unused classes like Workshop, WeaponEffect, Non-Common handlers, etc.
- Added proper loading dependenices, fixed hardcoded file separator in config, refactored some spelling mistakes.
- Fixed multiple enchants and weapons issues with offhand and added optional compatibility for RLCombat(BetterCombat)
- Improved some event priorities, removed receiveCanceled from events that shouldn't
- Added null checks for capabilities, air/empty checks for enchants, and some potential double/float/int casting issues
- Removed damage modifier from Range enchant as velocity affecting damage is not scaled linearly and causes conflicts with other mods/effects, lowered default velocity to 1.5 and added a config option to change it.
- Reworked Tunneling enchant to use a RayTrace, works seemingly just the same without the added tag handling, and added recursive check to prevent crashing/freezing.
- Reworked Stun effect to handle entity motion in CommonEventHandler instead of attempting to replace AI, handling similar to how Lycanite's handled the paralyze effect.
- Made Nunchaku capability actually countdown in LivingUpdateEvent, was not counting down previously.
- Added Flamed and Iced versions of dragonbone weapons, reworked InF material attributes to properly apply, reworked materials to directly reference the InF material registries if it is installed instead of attempting to re-register new materials.
- Reworked Spear to work properly, use Attributes for modifying Reach, moved Hammer/Axe/etc handling to CommonEventHandler and RLCombatCompatHandler for better compatability.
- Added descriptions to custom weapons to describe their effects.
- Fixed spelling errors in Lang, add pack.mcmeta for compatibility with new Forge resource loading.
- Added new animated textures for Flamed/Iced weapons (Made by ArtsyDy)
- Moved non-enchant/material config to use Forge's annotation system so it can be modified ingame easier.
- Integrated patches from RLTweaker and add config options, such as the ability to blacklist TileEntities from Tunneling, ignore air blocks when Tunneling, blacklist of mobs to affect with blindness, strength of blindness to affect mobs, etc.
- Fix improper == String check in config handler
- Fixed Switchbow compat checking config for loading alternate materials, should always run if its a Switchbow.
- Added compatibility to apply SoManyEnchantments enchants on custom weapons.
- Added compatibility to not bother registering custom cauldron handling if Inspirations is installed and has their own custom cauldron config enabled, instead register an Inspirations Cauldron Recipe to handle potions on swords instead.
- Added config for changing values related to potions on swords, like hits per dose, the ability to stack doses up to a cap, how much to divide the time by, a blacklist of disallowed potions, and a whitelist of weapon classes that potions can be applied on, instead of just ItemSword.
- Fixed tooltip rendering twice with potion effects, added the ability to clean off potions with a cauldron that has water and/or non-potion potion in it (Awkward, Thick, etc.)

### CODE REVIEW

**Comment by fonnymunkey** [comment]
Dec 10: Re-rework spears:
Fix spears getting voided, duped, and desynced.
Added compat checks for spears to prevent duping with InF Stone entities, and desyncing frozen effects.

**Comment by Block-Vader** [comment]
Thanks you very much for assisting me with my project. I am currently quite busy with scool, but I will look into this PR and merge it if possible when I will have time.
